glider_kinematics_gazebo_plugins/kinematics_control_plugins/src/direct_kinematics.cc

```diff
@@ -0,0 +1,131 @@
+#ifndef _DIRECT_KINEMATICS_HH_
+#define _DIRECT_KINEMATICS_HH_
+#define M_PI 3.14159265358979323846
+
+// defaults for the plugin
+#include <gazebo/gazebo.hh>
+#include <gazebo/physics/physics.hh>
+// for msg receiving for the plugin
+#include <gazebo/transport/transport.hh>
+#include <gazebo/msgs/msgs.hh>
+// for ros connection
+#include <thread>
+#include "ros/ros.h"
+#include "ros/callback_queue.h"
+#include "ros/subscribe_options.h"
+#include "std_msgs/Float32.h"
+#include "std_msgs/Float32MultiArray.h"
+#include "std_msgs/MultiArrayDimension.h"
+#include "gazebo_msgs/SetModelState.h"
+
+#include <cmath>
+#include <iostream>
+
+namespace gazebo
+{
+  /// \brief A plugin to control the vehicle kinematics directly
+  class DirectKinematics : public ModelPlugin
+  {
+    /// \brief Constructor
+  public:
+    DirectKinematics() {}
+
+    /// \brief The load function is called by Gazebo when the plugin is
+    /// inserted into simulation
+    /// \param[in] _model A pointer to the model that this plugin is
+    /// attached to.
+    /// \param[in] _sdf A pointer to the plugin's SDF element.
+  public:
+    virtual void Load(physics::ModelPtr _model, sdf::ElementPtr _sdf)
+    {
+      // Plugin load notifier
+      std::cerr << "\nThe Direct Kinematics plugin is attach to model[" << _model->GetName() << "]\n";
+
+      // Store the model pointer for convenience.
+      this->model = _model;
+      
+      // Default to zero velocity
+      ignition::math::Vector3d velocity = ignition::math::Vector3d(0, 0, 0);
+
+      // Check that the velocity element exists, then read the value
+      if (_sdf->HasElement("velocity"))
+        velocity = _sdf->Get<double>("velocity");
+
+      this->SetVelocity(velocity);
+      
+      //----------- FOR MSG CONNECTION ---------- //
+      // Create the node
+      this->node = transport::NodePtr(new transport::Node());
```

Comment:
I believe it would be more useful to have a ROS interface the plugin - instead of a Gazebo messaging interface.